airbyte-webapp/src/views/Connection/ConnectionForm/formConfig.tsx

```diff
@@ -77,19 +77,32 @@ export function useDefaultTransformation(): OperationCreate {
 export const connectionValidationSchema = yup
   .object({
     name: yup.string().required("form.empty.error"),
-    scheduleType: yup.string().oneOf([ConnectionScheduleType.manual, ConnectionScheduleType.basic]),
-    scheduleData: yup
-      .object({
-        basicSchedule: yup
+    scheduleType: yup
+      .string()
+      .oneOf([ConnectionScheduleType.manual, ConnectionScheduleType.basic, ConnectionScheduleType.cron]),
+    scheduleData: yup.mixed().when("scheduleType", (scheduleType) => {
+      if (scheduleType === ConnectionScheduleType.basic) {
+        return yup.object({
+          basicSchedule: yup
+            .object({
+              units: yup.number().required("form.empty.error"),
+              timeUnit: yup.string().required("form.empty.error"),
+            })
+            .defined("form.empty.error"),
+        });
+      } else if (scheduleType === ConnectionScheduleType.manual) {
+        return yup.mixed().notRequired();
+      }
+
+      return yup.object({
+        cron: yup
           .object({
-            units: yup.number().required("form.empty.error"),
-            timeUnit: yup.string().required("form.empty.error"),
+            cronExpression: yup.string().required("form.empty.error"),
```

Comment:
One quick thought: we could validate here that the `cronExpression` is a valid cron string by adding `.matches()` to this line, with a regex that validates it is a valid cron expression (can probably find a regex for this online). This would be nice as it would validate the cron expression whenever a user clicks away from the "Cron expression" input, and it would highlight the input red.

It seems like our backend API is doing some cron expression validations on the backend upon submit though, so this is probably not a blocking thing for this PR, but more of a "nice to have" if we could get it working.

Reply:
Hey, @lmossman I added util to test that. Instead of installing the package, I added that code to our codebase.